### PR TITLE
Monkey patch ncclient for issue 125

### PR DESCRIPTION
**Pull request by lindycoder**
This code applies the patch
https://github.com/ncclient/ncclient/pull/120

a test has been created to replicate the issue

This is a temporary fix for issue 125 while ncclient haven't merged and republished our fix

### CODE REVIEW

**Comment by fbouliane** [comment]
LGTM :+1:

File: tests/adapters/switches/juniper_issue_125_test.py [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+    def tearDown(self):
+        try:
+            with Timeout(seconds=3):
+                self.switch.rollback_transaction()
+                self.switch.end_transaction()
+                self.switch.disconnect()
+        except AssertionError:
+            pass
+
+    def test_my_stuff(self):
```
Comment (fbouliane) [inline tests/adapters/switches/juniper_issue_125_test.py]: More verbose method name ?

**Comment by fbouliane** [comment]
LGTM :+1:

**Comment by emmurd** [comment]
:+1:

**Pull request merged**